ui/src/app/settings/views/Dhcp/DhcpList/DhcpList.tsx

```diff
@@ -40,24 +51,28 @@ const getTargetName = (
     const machineItem = machines.find((machine) => machine.system_id === node);
     target = controllerItem || deviceItem || machineItem;
   }
-  return target && (target.name || target.hostname);
+  return (
+    target &&
+    (("name" in target && target.name) ||
```

Comment:
we should probably create some more identifier functions like `isNode` and `isDevice` etc (not in this PR though obvs)